### PR TITLE
GCW - 2473 request populate on first attempt

### PR DESCRIPTION
**Pull request by abulsayyad123**
Hi Team, 
This PR is for JIRA ticket GCW-GCW-2473- Request goods not populating on first attempt.(https://jira.crossroads.org.hk/browse/GCW-2473).  I have also added change for transport details on reflecting not reflecting after changing on confirm page. 
Please review the PR and let me know if any change is required.
Thanks.
Changes:
https://github.com/crossroads/stock.goodcity/pull/412/files#diff-530ad3f505e5fb3188441a432509c4f5R16
https://github.com/crossroads/stock.goodcity/pull/412/files#diff-530ad3f505e5fb3188441a432509c4f5R32
https://github.com/crossroads/stock.goodcity/pull/412/files#diff-34c3f50ad8fe4ce77a6341f846c73786R7
https://github.com/crossroads/stock.goodcity/pull/412/files#diff-272caf54bdda0689a6347736642ec50eR163
https://github.com/crossroads/stock.goodcity/pull/412/files#diff-272caf54bdda0689a6347736642ec50eR145

### CODE REVIEW

File: app/components/add-request.js [hunk 1]
```diff
@@ -1,43 +1,61 @@
...
+    }
   }),
 
   actions: {
     deleteRequest(reqId) {
       var i18n = this.get("i18n");
       this.get("messageBox").custom(
-        i18n.t("order.request.remove_req").string + this.get("model.code"),
-        i18n.t("order.request.remove").string, () => this.send("removeRequest", reqId),
+        i18n.t("order.request.remove_req").string + this.get("order.code"),
```
Comment (patrixr) [inline app/components/add-request.js]: just a thought, I think `i18n` supports string interpolation (not sure though).
It'd be worth testing if we can do something like:
```
I18n.translations = {
  en: {
    hello: 'hello {{name}}',
    }
  }
}

// hbs
{{ t 'hello' name=myName }}

```

File: app/models/designation.js [hunk 2]
```diff
@@ -135,46 +142,56 @@ export default Model.extend({
     }
   }),
 
-  appointmentTransport: Ember.computed("orderTransport", function() {
-    let i18n = this.get("i18n");
-    return this.get("orderTransport.transportType") === "self" ?
-      i18n.t("order.appointment.self_vehicle") : i18n.t("order.appointment.hire_vehicle");
-  }),
+  appointmentTransport: Ember.computed(
```
Comment (patrixr) [inline app/models/designation.js]: Unrelated to your changes, but I'm thinking we should rename properties which are translations, to something like `appointmentTransportLabel`, to make it crystal clear that it's a display string

Comment (abulsayyad123) [reply app/models/designation.js]: Sure will do.

File: app/components/add-request.js (hunk 1 above)
Comment (Darshan-Chauhan) [reply app/components/add-request.js]: @patrixr , yes it does support it.

CC: @abulsayyad123

Comment (abulsayyad123) [reply app/components/add-request.js]: Yes it is possible, and it has implemented where we need to pass attributes to translations. In this case, message-box is used at many places which require to add changes at many place. I think we can take away this point for future reference.

Comment (Darshan-Chauhan) [reply app/components/add-request.js]: It's not change in structure of messageBox.

**Pull request merged**